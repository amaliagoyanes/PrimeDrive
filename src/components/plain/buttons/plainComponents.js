import React from 'react';
import {
    TouchableOpacity,
    Text,
    View,
    TextInput,
    Image,
} from 'react-native';
import {connect} from 'react-redux';
import {injectIntl, intlShape, FormattedMessage} from 'react-intl';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {Col, Row} from 'react-native-easy-grid';
import styles from '../../../style/styles';
import {scale, scaleText} from '../../../scaling';

export class IconButton extends React.Component {
    onPress() {
        if (this.props.measures)
            this._btn.measure((fx, fy, width, height, px, py) => {
                this.props.onPress(px, py, width, height, fx, fy);
            });
        else
            this.props.onPress();
    }

    render() {
        let props = this.props;
        let text = this.props.price;
        if (this.props.intl !== undefined && this.props.intl.messages !== undefined && this.props.intl.messages[this.props.price])
            text = this.props.intl.messages[this.props.price];
        return (
            <Col>
                <TouchableOpacity
                    onLongPress={() => {
                        console.log('this.props.onLongPress');
                        if (this.props.onLongPress)
                            this.props.onLongPress();
                    }} onPress={() => this.onPress()} ref={elt => this._btn = elt}
                    style={props.active ? [styles.button, styles.activeColor, props.style] : [styles.button, props.style]}>
                    {props.image !== undefined &&
                    <Image source={{uri: props.image}} style={[styles.imageOnButton]}></Image>
                    }
                    {props.badge !== undefined &&
                    <View style={styles.iconButtonBadge}>
                        <Text style={[styles.text, {fontSize: scaleText(10)}]}>{props.badge}</Text>
                    </View>
                    }
                    {props.status !== undefined &&
                    <View style={[styles.iconButtonBadge, {backgroundColor: props.status ? '#2dab61' : '#707076'}]}/>
                    }
                    {props.icon !== undefined &&
                    <Icon size={scaleText(18)} name={props.icon} color="white"/>}
                    {props.text !== undefined &&
                    <Text
                        style={[styles.text, {
                            textAlign: 'center',
                            fontSize: scaleText(12)
                        }]}>{props.text.toUpperCase()}</Text>}
                    {props.price !== undefined &&
                    <Text style={[styles.text, {
                        textAlign: 'center',
                        fontSize: scaleText(10)
                    }]}>{text.toUpperCase()}</Text>}
                </TouchableOpacity>
            </Col>
        )
    }
}

export class ButtonItem extends React.Component {
    render() {
        let props = this.props;
        let text = this.props.text;
        if (this.props.intl !== undefined && this.props.intl.messages !== undefined && this.props.intl.messages[this.props.text])
            text = this.props.intl.messages[this.props.text];
        return (
            <Col>
                <TouchableOpacity onPress={props.onPress}
                                  activeOpacity={props.disabled !== undefined && props.disabled ? 1 : 0.2}
                                  style={props.active ? [styles.button, styles.activeColor, props.style] : [styles.button, props.style]}>
                    <Text style={[styles.text, props.textStyle]}>{text.toUpperCase()}</Text>
                    {props.detailsText &&
                    <Text style={[styles.text, props.textStyle, {textAlign: 'right'}]}>{props.detailsText}</Text>
                    }
                    {props.modalActive &&
                    <View
                        style={[styles.modalActiveArrow, {backgroundColor: props.style[props.style.length - 1] ? props.style[props.style.length - 1].backgroundColor : props.active ? '#6f7076' : '#53535a'}]}/>
                    }
                </TouchableOpacity>
            </Col>
        )
    }
}

export class TwoTextButtonItem extends React.Component {
    render() {
        let props = this.props;
        let text = this.props.text;
        if (this.props.intl !== undefined && this.props.intl.messages !== undefined && this.props.intl.messages[this.props.text])
            text = this.props.intl.messages[this.props.text];
        return (
            <Col>
                <TouchableOpacity onPress={props.onPress}
                                  activeOpacity={props.disabled !== undefined && props.disabled ? 1 : 0.2}
                                  style={props.active ? [styles.button, styles.activeColor, props.style] : [styles.button, props.style]}>
                    <Row>
                        <Col style={{width: '70%', marginLeft: scale(10), justifyContent: 'center',}}>
                            <Text
                                style={[styles.text, props.textStyle, {textAlign: 'left'}]}>{text.toUpperCase()}</Text>
                        </Col>
                        {props.detailsText &&
                        <Col style={{marginRight: scale(10), justifyContent: 'center',}}>
                            <Text
                                style={[styles.text, props.textStyle, {textAlign: 'right'}]}>{props.detailsText}</Text>
                        </Col>
                        }
                    </Row>
                </TouchableOpacity>
            </Col>
        )
    }
}


export class ButtonMeasures extends React.Component {
    onPress() {
        this._btn.measure((fx, fy, width, height, px, py) => {
            this.props.onLongPress(px, py);
        });
    }

    render() {
        let props = this.props;
        let text = this.props.text;
        if (this.props.intl !== undefined && this.props.intl.messages !== undefined && this.props.intl.messages[this.props.text])
            text = this.props.intl.messages[this.props.text];
        return (
            <Col>
                <TouchableOpacity onLongPress={() => this.onPress()} ref={elt => this._btn = elt}
                                  onPressIn={() => {
                                      this.lastTimePressIn = new Date().getTime();
                                  }}
                                  onPressOut={() => {
                                      let time = new Date().getTime();
                                      let diff = time - this.lastTimePressIn;
                                      if (diff < 500) {
                                          if (this.props.onPress !== undefined && this.props.onDoublePress === undefined)
                                              this.props.onPress();
                                          else if (this.lastTimePress && time - this.lastTimePress < 1500) {
                                              if (this.props.onDoublePress !== undefined)
                                                  this.props.onDoublePress();
                                              delete this.lastTimePress;
                                          } else {
                                              this.lastTimePress = time;
                                          }
                                      }
                                  }}
                                  style={props.active ? [styles.button, styles.activeColor, props.style] : [styles.button, props.style]}>
                    <Text style={[styles.text, props.textStyle]}>{text.toUpperCase()}</Text>
                    {(props.detailsText &&
                        <Text style={[styles.text, props.textStyle, {textAlign: 'right'}]}>{props.detailsText}</Text>
                    )}
                </TouchableOpacity>
            </Col>
        );
    }
}

export const TableItem = (props) => (
    <Col>
        <TouchableOpacity onPress={props.onPress}
                          style={[styles.button, styles.tableItem, props.style]}>
            <Row style={{alignItems: 'center', justifyContent: 'center'}}>
                <Text style={[styles.text, styles.tableItemNum, props.textStyle]}>{props.num}</Text>
            </Row>

            <Row style={{alignItems: 'center', justifyContent: 'center'}}>
                {(props.time &&
                    <Text
                        style={[styles.text, {
                            fontSize: scaleText(10),
                            color: '#fefefe'
                        }, props.textStyle]}>{props.time}</Text>
                )}
            </Row>
        </TouchableOpacity>
    </Col>
);
