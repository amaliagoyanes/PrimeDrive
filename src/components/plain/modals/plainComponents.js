import moment from 'moment';
import React from 'react';
import {
    TouchableOpacity,
    Text,
    View,
    TextInput
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import {Col, Row} from 'react-native-easy-grid';
import styles from '../../../style/styles';
import {scale, scaleText} from '../../../scaling';
import {Timer} from '../index.js';


export const ModalCloseButton = (props) => (
    <Row size={1}>
        <Col size={20}/>
        <Col size={1}>
            <TouchableOpacity onPress={props.close}
                              style={[styles.bigTimesIcon, {
                                  marginTop: 0, marginLeft: scale(32), padding: scale(5),
                                  backgroundColor: 'transparent'
                              }]}>

                <Icon size={scaleText(32)} name="close" color="#313138"/>
            </TouchableOpacity>
        </Col>
    </Row>
);

export class ModalHeader extends React.Component {
    constructor(props) {
        super(props);

        this.getToday = this.getToday.bind(this);
    }


    getToday() {
        return moment().format('DD.MM.YYYY');
    }

    render() {
        let props = this.props;
        return (
            <Row size={1} style={styles.modalHeader}>
                <Col size={1}>
                    <Timer style={{
                        fontSize: scaleText(12), color: '#313138', textAlign: 'left'
                    }}/>
                </Col>
                <Col size={1}>
                    <Text style={[styles.text, {
                        fontSize: scaleText(14), color: '#313138', textAlign: 'center'
                    }]}>{props.intl !== undefined && props.intl.messages[props.modalName]} </Text>
                </Col>
                <Col size={1}>
                    <Text style={[styles.text, {
                        fontSize: scaleText(12),
                        color: '#313138',
                        textAlign: 'right'
                    }]}>{this.getToday()}</Text>
                </Col>
            </Row>
        )
    }
}
