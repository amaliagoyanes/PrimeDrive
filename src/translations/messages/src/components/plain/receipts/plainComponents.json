[
  {
    "id": "app.totalUpper",
    "defaultMessage": "TOTAL"
  },
  {
    "id": "app.paid",
    "defaultMessage": "Paid"
  }
]