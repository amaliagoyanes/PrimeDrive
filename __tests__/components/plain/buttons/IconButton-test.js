'use strict';

import 'react-native';
import React from 'react';
import {IconButton} from '../../../../src/components/plain/buttons/plainComponents';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <IconButton text="Button" onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly active', () => {
    const tree = renderer.create(
        <IconButton text="Button" active onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with badge', () => {
    const tree = renderer.create(
         <IconButton text="Button" onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly active with badge', () => {
    const tree = renderer.create(
         <IconButton text="Button" active onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with badge and status', () => {
    const tree = renderer.create(
         <IconButton text="Button" status onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly active with badge and status', () => {
    const tree = renderer.create(
         <IconButton text="Button" status active onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with badge, status, price', () => {
    const tree = renderer.create(
         <IconButton text="Button" price="10.55" status onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly active with badge, status, price', () => {
    const tree = renderer.create(
         <IconButton text="Button" price="10.55" status active onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with badge and price', () => {
    const tree = renderer.create(
         <IconButton text="Button" price="10.55" onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly active with badge and price', () => {
    const tree = renderer.create(
         <IconButton text="Button" price="10.55" active onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly active with badge, status, price, icon', () => {
    const tree = renderer.create(
         <IconButton text="Button" price="10.55" icon="access-point" status active onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with badge, status, price, icon', () => {
    const tree = renderer.create(
         <IconButton text="Button" price="10.55" icon="access-point" status onPress={jest.fn} />).toJSON();
    expect(tree).toMatchSnapshot();
});

