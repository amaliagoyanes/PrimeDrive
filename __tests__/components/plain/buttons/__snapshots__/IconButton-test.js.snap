// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#53535a",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
  </View>
</View>
`;

exports[`renders correctly active 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6f7076",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
  </View>
</View>
`;

exports[`renders correctly active with badge 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6f7076",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
  </View>
</View>
`;

exports[`renders correctly active with badge and price 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6f7076",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      10.55
    </Text>
  </View>
</View>
`;

exports[`renders correctly active with badge and status 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6f7076",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#707076",
            "height": 20,
            "justifyContent": "center",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 20,
          },
          Object {
            "backgroundColor": "#2dab61",
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
  </View>
</View>
`;

exports[`renders correctly active with badge, status, price 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6f7076",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#707076",
            "height": 20,
            "justifyContent": "center",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 20,
          },
          Object {
            "backgroundColor": "#2dab61",
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      10.55
    </Text>
  </View>
</View>
`;

exports[`renders correctly active with badge, status, price, icon 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#6f7076",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#707076",
            "height": 20,
            "justifyContent": "center",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 20,
          },
          Object {
            "backgroundColor": "#2dab61",
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      10.55
    </Text>
  </View>
</View>
`;

exports[`renders correctly with badge 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#53535a",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
  </View>
</View>
`;

exports[`renders correctly with badge and price 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#53535a",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      10.55
    </Text>
  </View>
</View>
`;

exports[`renders correctly with badge and status 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#53535a",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#707076",
            "height": 20,
            "justifyContent": "center",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 20,
          },
          Object {
            "backgroundColor": "#2dab61",
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
  </View>
</View>
`;

exports[`renders correctly with badge, status, price 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#53535a",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#707076",
            "height": 20,
            "justifyContent": "center",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 20,
          },
          Object {
            "backgroundColor": "#2dab61",
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      10.55
    </Text>
  </View>
</View>
`;

exports[`renders correctly with badge, status, price, icon 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#53535a",
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginLeft": 5,
        "marginRight": 5,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "#707076",
            "height": 20,
            "justifyContent": "center",
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 20,
          },
          Object {
            "backgroundColor": "#2dab61",
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 18,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }
    >
      BUTTON
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Gotham-Book",
            "fontSize": 14,
          },
          Object {
            "fontSize": 10,
            "textAlign": "center",
          },
        ]
      }
    >
      10.55
    </Text>
  </View>
</View>
`;
